modules/cardano.nix

```diff
@@ -0,0 +1,114 @@
+# XXX: rename this file:  cardano-node-config vs. cardano-nodes-config is CRUEL
+with (import ./../lib.nix);
+
+globals: imports: params:
+  { pkgs, nodes, config, resources, options, ...}:
+  let
+    nodeNameToPublicIP   = name: let ip = nodes.${name}.config.services.cardano-node.publicIP;
+                                 in if ip == null then "" else ip;
+    neighbourPairs       = map (n: { name = n; ip = nodeNameToPublicIP n; })
+                               (builtins.trace "${params.name}: role '${params.nodeType}'" params.peers);
+    ppNeighbour          = n: "${n.name}: ${n.ip}";
+    sep                  = ", ";
+    cfg                  = config.services.cardano-node;
+  in {
+    imports = [
+      ./common.nix
+      ./amazon-base.nix
+      ./network-wide.nix
+      (import ./cardano-service.nix globals params)
+    ] ++ map (path: import path globals params) imports;
+
+    global.organisation = params.org;
+
+    services.dnsmasq.enable = true;
+    services.dnsmasq.servers = [ "127.0.0.1" ];
+
+    # TODO: DEVOPS-8
+    #deployment.ec2.ami = (import ./amis.nix).${config.deployment.ec2.region};
+    deployment.ec2.region         = mkForce params.region;
+    deployment.ec2.zone           = mkForce params.zone;
+    deployment.ec2.accessKeyId    = params.accessKeyId;
+    deployment.ec2.keyPair        = resources.ec2KeyPairs.${params.keyPairName};
+    deployment.ec2.securityGroups =
+      with params;
+      let sgNames =
+           optionals typeIsExplorer              [ "allow-to-explorer-${params.region}" ]
+        ++ optionals typeIsCore                  [ "allow-cardano-static-peers-${params.name}-${params.region}-${params.org}" ]
+        ++ optionals typeIsRelay                 [ "allow-kademlia-public-udp-${params.region}"
+                                                   "allow-cardano-public-tcp-${params.region}" ]
+        ++ optionals config.global.enableEkgWeb  [ "allow-ekg-public-tcp-${params.region}-${params.org}" ];
+      in map (resolveSGName resources)
+             (if config.global.omitDetailedSecurityGroups
+              then [ "allow-all-${params.region}-${params.org}" ]
+              else sgNames);
```

Comment:
This implementation should be reconsidered.  Perhaps as part of https://issues.serokell.io/issue/DEVOPS-460

We don't really want environment/component sensitivity hard-coded here.